syntax = "proto3";
package shapleq.proto;

enum SessionType {
    ADMIN = 0;
    PUBLISHER = 1;
    SUBSCRIBER = 2;
}

message Group {
    uint64 group_id = 1;
    uint64 owner_id = 2;
    string group_name = 3;
}

message Partition {
    uint64 partition_id = 1;
    uint64 offset = 2;
}

message Topic {
    string name = 1;
    string description = 2;
    uint32 num_partitions = 3;
    uint32 replication_factor = 4;
    uint64 last_offset = 5;
}